editor/src/core/shared/jsx-attributes.ts

```diff
@@ -414,6 +414,9 @@ export function deeplyCreatedValue(path: PropertyPath, value: JSXAttribute): JSX
   }, value)
 }
 
+function isArrayIndex(maybeIndex: string | number): maybeIndex is number {
+  return !isNaN(maybeIndex as any)
```

Comment:
We have a function `parseNumber` in `math-utils.ts` that is similar, but calls `parseFloat` first. It seems like that would be a safer option here.